### PR TITLE
fix: prometheus can't scrape serviceMonitor job

### LINKED ISSUE #30594: Prometheus can't scrape serviceMonitor job
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What happened?

Hello , I am following the instructions on the docs https://docs.cilium.io/en/stable/observability/metrics/#cilium-metrics

When I install cilium and trying to enable Monitoring & Metric, instead of scrape_configs, I found that serviceMonitor can't be scaped by prometheus. 

Here's my install command
`helm upgrade --install cilium cilium/cilium --version 1.14.5 --namespace kube-system -f values.yaml`

Here's my values.yaml
```yaml
hubble:
  enabled: true

  relay:
    enabled: true
    prometheus:
      enabled: true
      serviceMonitor:
        enabled: true

  metrics:
    serviceMonitor:
      enabled: true

    enableOpenMetrics: true
    enabled:
      - dns
      - drop
      - tcp
      - flows-to-world
      - port-distribution
      - icmp
      - "flow:sourceContext=workload-name|reserved-identity;destinationContext=workload-name|reserved-identity"
      - "httpV2:exemplars=true;labelsContext=source_ip,source_namespace,source_workload,destination_ip,destination_namespace,destination_workload,traffic_direction;sourceContext=workload-name|reserved-identity;destinationContext=workload-name|reserved-identity"

prometheus:
  enabled: true
  serviceMonitor:
    enabled: true

operator:
  enabled: true
  prometheus:
    enabled: true
    serviceMonitor:
      enabled: true
```

### Cilium Version

1.14.0

### Kernel Version

5.15.0-91-generic

### Kubernetes Version

1.27

### Sysdump

_No response_

### Relevant log output

_No response_

### Anything else?

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by HsiuyaTsao**
Fixes: #30594 
Prometheus can't scrape serviceMonitor job, plz see PR commit

### CODE REVIEW

File: install/kubernetes/cilium/templates/cilium-agent/servicemonitor.yaml [hunk 1]
```diff
@@ -7,6 +7,7 @@ metadata:
   namespace: {{ .Values.prometheus.serviceMonitor.namespace | default .Release.Namespace }}
   labels:
     app.kubernetes.io/part-of: cilium
+    release: prometheus
```
Comment (sayboras) [inline install/kubernetes/cilium/templates/cilium-agent/servicemonitor.yaml:10]: is this label a must ? or alternatively can it be configured via .Values.prometheus.serviceMonitor.labels?

**Comment by chancez** [comment]
As @sayboras said, this doesn't seem essential and can be specified in the extra labels fields available, or you could even adjust your Prometheus configuration to require different labels. Gonna be a NACK here.

File: install/kubernetes/cilium/templates/cilium-agent/servicemonitor.yaml (hunk 1 above)
Comment (HsiuyaTsao) [reply install/kubernetes/cilium/templates/cilium-agent/servicemonitor.yaml:10]: Hi @sayboras, 

I have found that there is a similiar related issue in prometheus-community. https://github.com/prometheus-community/helm-charts/issues/1911. It seems adding label is not a must, the issue can be solved by adding prometheus.prometheusSpec.serviceMonitorSelectorNilUsesHelmValues: false.

Thanks for your reply.

Comment (sayboras) [reply install/kubernetes/cilium/templates/cilium-agent/servicemonitor.yaml:10]: Seems like we can close this PR, feel free to let us know if you have any question/concern in Cilium slack.

Looking forward to your next contribution, thanks.

**Pull request closed without merging**